cmd/skopeo/utils.go

```diff
@@ -9,12 +9,20 @@ import (
 	"github.com/urfave/cli"
 )
 
-func contextFromGlobalOptions(c *cli.Context, credsFlag string) (*types.SystemContext, error) {
+func contextFromGlobalOptions(c *cli.Context, credsFlag, certPathFlag, tlsVerifyFlag string) (*types.SystemContext, error) {
 	ctx := &types.SystemContext{
-		RegistriesDirPath:           c.GlobalString("registries.d"),
+		RegistriesDirPath: c.GlobalString("registries.d"),
+		// DEPRECATED: keep this here for backward compatibility, but override
+		// them if per subcommand flags are provided (see below).
```

Comment:
> that means that if you use `skopeo copy` with just `--tls-verify` it'll be applied to both src and dest,

That’s ugly, sure, but if this PR is already adding a warning about the flag being deprecated, it seems acceptable to me.